packages/python/accern_data/accern_data.py

```diff
@@ -348,17 +404,26 @@ def stringify_dates(obj: Dict[str, Any]) -> Dict[str, Any]:
 
     def split(
             self,
-            batch: Union[List[pd.DataFrame], List[Dict[str, Any]]],
+            batch: List[Dict[str, Any]],
             value: pd.Timestamp) -> List[Dict[str, Any]]:
         result = []
         for record in batch:
-            assert isinstance(record, dict)
             if record["harvested_at"] <= value:
                 result.append(record)
         return result
 
+    def iterate_data(
+            self,
+            data: Optional[List[Dict[str, Any]]],
+            indicator: ProgressIndicator,
+            chunk_size: Optional[int] = None) -> Iterator[Dict[str, Any]]:
+        if data is not None:
+            for rec in data:
+                indicator.update(1)
+                yield rec
+
 
-class DataClient():
+class DataClient:
     def __init__(
             self,
             url: str,
```

Comment:
have a main indicator object attribute like with mode. if you pass none here you use the default if you pass none in the function calls you use the attribute